static/index.js

```diff
@@ -14,23 +9,9 @@ function formatCrateName(crateTitleAndVersion) {
             clipboard.innerHTML = resetClipboardIcon;
         }
 
-        function copyTextHandler() {
-            const crateTitleAndVersion = document.getElementById("crate-title");
-            // On rustdoc pages, we use `textTransform: uppercase`, which copies as uppercase.
-            // To avoid that, reset the styles temporarily.
-            const oldTransform = crateTitleAndVersion.style.textTransform;
-            crateTitleAndVersion.style.textTransform = "none";
-            const temporaryInput = document.createElement("input");
-
-            temporaryInput.type = "text";
-            temporaryInput.value = formatCrateName(crateTitleAndVersion.innerText);
-
-            document.body.append(temporaryInput);
-            temporaryInput.select();
-            document.execCommand("copy");
-
-            temporaryInput.remove();
-            crateTitleAndVersion.style.textTransform = oldTransform;
+        async function copyTextHandler() {
+            const metadata = JSON.parse(document.getElementById("crate-metadata").innerText)
+            await navigator.clipboard.writeText(`${metadata.name} = "${metadata.version}"`)
```

Comment:
Ah, one thing I should note is that `navigator.clipboard` requires a secure context, so this won't work when running on an http-only dev instance (I have my dev instance behind a reverse proxy adding https to it so was able to verify it works, on firefox at least).

Reply:
So, you can actually cheat here to do that (we do it in rustdoc for example):

```js
const el = document.createElement("textarea");
el.value = `${metadata.name} = "${metadata.version}"`;
el.setAttribute("readonly", "");
// To not make it appear on the screen.
el.style.position = "absolute";
el.style.left = "-9999px";

document.body.appendChild(el);
el.select();
document.execCommand("copy");
document.body.removeChild(el);
```

No need for https anymore. :)

Reply:
Oh that's kinda what we were doing before in fact.

Reply:
Yeah, I sort of wanted to get away from that unspecced API to the official replacement, and for docs.rs I don't think the insecure-context usecase is very important.

Reply:
But for testing locally, it is. I think we should keep the "unspecced API" (which in fact is, it's just a round-about way to do the same) for making it easier to test changes locally.

Reply:
Reverted this part of the change.